e2e/claims.service.e2e.ts

```diff
@@ -825,109 +825,103 @@ describe('Сlaim tests', () => {
+      test('resolveCredentialAndVerify should resolve a Verifiable Credential and call correct verification method (verifyVC)', async () => {
+        const roleName = `${resolveVC}.${root}`;
+        await signerService.connect(rootOwner, ProviderType.PrivateKey);
+        await signerService.connect(staticIssuer, ProviderType.PrivateKey);
+        await createExampleSignedCredential([], roleName);
+        const { issuedToken } = await enrolAndIssue(rootOwner, staticIssuer, {
+          subjectDID: rootOwnerDID,
+          claimType: roleName,
+          registrationTypes: [
+            RegistrationTypes.OnChain,
+            RegistrationTypes.OffChain, // role type issuer should have offchain claim
+          ],
+          publishOnChain: true,
+          issuerFields: [],
+        });
+        await signerService.connect(rootOwner, ProviderType.PrivateKey);
+        let subjectDoc = await didRegistry.getDidDocument({
+          did: rootOwnerDID,
+          includeClaims: true,
+        });
+        mockCachedDocument.mockResolvedValueOnce(subjectDoc);
 
-      subjectDoc = await didRegistry.getDidDocument({
-        did: rootOwnerDID,
-        includeClaims: true,
-      });
-      const result = await claimsService.resolveCredentialAndVerify(
-        rootOwnerDID,
-        roleName
-      );
-      expect(result.errors).toHaveLength(0);
-      expect(result.isVerified).toBeTruthy;
-      expect(claimsService.verifyVc).toHaveBeenCalled;
-    });
-    test('resolveAndVerify should resolve an EIP191JWT and verify the claim', async () => {
-      const roleName = `${verifyOffChainClaimRole}.${root}`;
-      await domainsService.createRole({
-        roleName: verifyOffChainClaimRole,
-        namespace,
-        data: roles[roleName],
-        returnSteps: false,
-      });
+        await claimsService.publishPublicClaim({
+          claim: { token: issuedToken },
+        });
 
-      await signerService.connect(staticIssuer, ProviderType.PrivateKey);
-      const { issuedToken } = await enrolAndIssue(rootOwner, staticIssuer, {
-        subjectDID: rootOwnerDID,
-        claimType: roleName,
-        registrationTypes: [
-          RegistrationTypes.OnChain,
-          RegistrationTypes.OffChain,
-        ],
-        publishOnChain: true,
-        issuerFields: [],
-      });
-      await signerService.connect(rootOwner, ProviderType.PrivateKey);
-      let subjectDoc = await didRegistry.getDidDocument({
-        did: rootOwnerDID,
-        includeClaims: true,
-      });
-      mockCachedDocument.mockResolvedValueOnce(subjectDoc);
+        subjectDoc = await didRegistry.getDidDocument({
+          did: rootOwnerDID,
+          includeClaims: true,
+        });
+        const result = await claimsService.resolveCredentialAndVerify(
+          rootOwnerDID,
+          roleName
+        );
+        expect(result.errors).toHaveLength(0);
+        expect(result.isVerified).toBeTruthy;
+        expect(claimsService.verifyVc).toHaveBeenCalled;
```

Comment:
Best not to test implementation details, IMO. (Unless there is a good reason in this case)

Reply:
Yes, I mean the use of `toHaveBeenCalled`. In my opinion, rather than testing if a specific function has been called, I think that it is better to have tests that confirm the effect of the function. The way to do that, in my opinion, is to test before the positive and negative cases. So this test is `resolveCredentialAndVerify should resolve a Verifiable Credential and call correct verification method (verifyVC)`, so I would change to:
- `resolveCredentialAndVerify should resolve and verify a Verifiable Credential` -> a rename of this test
- `resolveCredentialAndVerify should throw error if <some case that verfiyVC covers>` (maybe you have made this test already)